src/devices/dspic33e.cpp

```diff
@@ -18,411 +18,422 @@
-#define DELAY_P12_PIC24FJ	25000	// 25ms
-#define DELAY_P13_DSPIC33E	1600	// 1.6ms
-#define DELAY_P13_PIC24FJ	20		// 20us
-#define DELAY_P14			1		// 1us MAX!
-#define DELAY_P15			1		// 10ns
-#define DELAY_P16			0		// 0s
-#define DELAY_P17   		0		// 0s - 100ns MAX!
-#define DELAY_P18			1000	// 1ms
-#define DELAY_P19			1		// 25ns
-#define DELAY_P20			25000	// 25ms
-#define DELAY_P21			1		// 1us - 500us MAX!
-
-#define ENTER_PROGRAM_KEY	0x4D434851
+#define DELAY_P1           1      // 200ns
+#define DELAY_P1A          1      // 80ns
+#define DELAY_P1B          1      // 80ns
+#define DELAY_P2           1      // 15ns
+#define DELAY_P3           1      // 15ns
+#define DELAY_P4           1      // 40ns
+#define DELAY_P4A          1      // 40ns
+#define DELAY_P5           1      // 20ns
+#define DELAY_P6           1      // 100ns
+#define DELAY_P7_DSPIC33E  25000  // 25ms
+#define DELAY_P7_PIC24FJ   50000  // 50ms
+#define DELAY_P8           12     // 12us
+#define DELAY_P9A          10     // 10us
+#define DELAY_P9B          15     // 15us - 23us max!
+#define DELAY_P10          1      // 400ns
+#define DELAY_P11_DSPIC33E 116000 // 116ms
+#define DELAY_P11_PIC24FJ  25000  // 25ms
+#define DELAY_P12_DSPIC33E 23000  // 23ms
+#define DELAY_P12_PIC24FJ  25000  // 25ms
+#define DELAY_P13_DSPIC33E 1600   // 1.6ms
+#define DELAY_P13_PIC24FJ  20     // 20us
+#define DELAY_P14          1      // 1us MAX!
+#define DELAY_P15          1      // 10ns
+#define DELAY_P16          0      // 0s
+#define DELAY_P17          0      // 0s - 100ns MAX!
+#define DELAY_P18          1000   // 1ms
+#define DELAY_P19          1      // 25ns
+#define DELAY_P20          25000  // 25ms
+#define DELAY_P21          1      // 1us - 500us MAX!
+
+#define ENTER_PROGRAM_KEY 0x4D434851
 
 #define reset_pc() send_cmd(0x040200)
 #define send_nop() send_cmd(0x000000)
 
-static unsigned int counter=0;
+static unsigned int counter = 0;
 static uint16_t nvmcon;
 
 /* Send a 24-bit command to the PIC (LSB first) through a SIX instruction */
 void dspic33e::send_cmd(uint32_t cmd)
 {
-	uint8_t i;
-
-	GPIO_CLR(pic_data);
-
-	/* send the SIX = 0x0000 instruction */
-	for (i = 0; i < 4; i++) {
-		GPIO_SET(pic_clk);
-		delay_us(DELAY_P1B);
-		GPIO_CLR(pic_clk);
-		delay_us(DELAY_P1A);
-	}
-
-	delay_us(DELAY_P4);
-
-	/* send the 24-bit command */
-	for (i = 0; i < 24; i++) {
-		if ( (cmd >> i) & 0x00000001 )
-			GPIO_SET(pic_data);
-		else
-			GPIO_CLR(pic_data);
-		delay_us(DELAY_P1A);
-		GPIO_SET(pic_clk);
-		delay_us(DELAY_P1B);
-		GPIO_CLR(pic_clk);
-	}
-
-	delay_us(DELAY_P4A);
-
+    uint8_t i;
+
+    GPIO_CLR(pic_data);
+
+    /* send the SIX = 0x0000 instruction */
+    for (i = 0; i < 4; i++)
+    {
+        GPIO_SET(pic_clk);
+        delay_us(DELAY_P1B);
+        GPIO_CLR(pic_clk);
+        delay_us(DELAY_P1A);
+    }
+
+    delay_us(DELAY_P4);
+
+    /* send the 24-bit command */
+    for (i = 0; i < 24; i++)
+    {
+        if ((cmd >> i) & 0x00000001)
```

Comment:
The goal was just to distill the diff between the dspic33f and dspicc33e implementations by eliminating formatting differences - I've yet to decide what I'm doing with it. On one hand, this is a quick, cheap way to be able to flash the dsPIC33CH from a Pi for primitive HIL purposes, so I wouldn't need to research / request a purchase of a x86/x64 SBC of some kind to run Microchip's junk but on the other hand it would be super-tightly coupled to vendor, platform, etc., so the possibility for much reuse is very low. I hadn't planned to upstream the changes and I'm probably just going to ditch this as not a good investment. Too many cons.